dev/TEAL.md

```diff
@@ -49,6 +58,30 @@ Constants are loaded into the environment by two opcodes, `intcblock` and `bytec
 
 Constants are pushed onto the stack by `intc`, `intc_[0123]`, `bytec`, and `bytec_[0123]`. The assembler will handle converting `int N` or `byte N` into the appropriate form of the instruction needed.
 
+### Named Integer Constants
+
+#### OnComplete
+| Value | Constant name | Description |
+| --- | --- | --- |
+| 0 | NoOp | Application transaction will simply call its ApprovalProgram. |
+| 1 | OptIn | Application transaction will allocate some LocalState for the application in the sender's account. |
```

Comment:
Replace the following with the corresponding text from `ledger.md`:
  - `OptInOC` (value `1`): Before executing the `ApprovalProgram`, allocate
    local state for this application into the sender's account data.
  - `CloseOutOC` (value `2`): After executing the `ApprovalProgram`, clear any
    local state for this application out of the sender's account data.
  - `ClearStateOC` (value `3`): Don't execute the `ApprovalProgram`, and instead
    execute the `ClearStateProgram` (which may not reject this transaction).
    Additionally, clear any local state for this application out of the sender's
    account data as in `CloseOutOC`.
  - `UpdateApplicationOC` (value `4`): After executing the `ApprovalProgram`,
    replace the `ApprovalProgram` and `ClearStateProgram` associated with this
    application ID with the programs specified in this transaction.
  - `DeleteApplicationOC` (value `5`): After executing the `ApprovalProgram`,
    delete the application parameters from the account data of the application's
    creator.